plugin/manager.go

```diff
@@ -129,6 +136,16 @@ func (m *manager) InitWithOptions(options Options) (*synccontext.RegisterContext
 	ctrl.SetLogger(logger)
 	ctx := klog.NewContext(context.Background(), logger)
 
+	go func() {
```

Comment:
Since you use m.interceptors here in a goroutine and then below as well, we need some mutex around it